### ISSUE TITLE
Can you share some settings for training set and other parameters?

### ISSUE THREAD

**Issue by tae1yoon** [hamed-zamani/snrm#9]
Hi, 
I'm quite a big fan for your research papers, I think your "Relevance based word embedding" and other papers are very interesting.  
As I'm also interested in your SNRM paper, so I tried to used your SNRM model based on my own data set.  
However, I have encountered two problems.  

First of all, in my own dataset the retrieval performance of SNRM is very low.  
For example, precision@20 is almost 0 for every queries in my test collection.
Second, the number of latent terms showed a different tendency compared to your paper.   

| Number of Words |Average number of Lterms | Median number of Lterms |
|------------------|------------------|-------------------|
| 1|34.69 | 32.5 |
| 2|37.87 | 37.0 |
| 3|38.81 | 38.0 |
| 4|38.79 | 37.0 |
| 5|39.46 | 39.0 |
| 6|39.36 | 39.0 |
| 7|39.1 | 37.5 |
| 8|39.27 | 39.0 |
| 9|39.56044 | 39.0 |
| 10|40.55 | 39.0 |
| 5 ~ 34 | 47.49 | 47|
| 145 ~ 204| 71.12 | 70|
| 250 ~ 440| 85.91 | 87|
| 1000| 103.46| 105 |

As in your paper, the number of latent terms for the query is below 5 and the number of latent terms for the document is around 130~. 

For your information, I attach the parameter setting and how I constructed training set below.
I constructed train set based on set of (query, 1st ranking document, 2~9th ranking document).  
I set the parameters as   
```
'n-gram': 3
'emb_dim': 128
'learning_rate': 0.0001
'dropout_parameter': 1.0
'regularization_term': 0.0001
'hidden_1': 500
'hidden_2': 100
'hidden_3': 10000
'max_q_len': 10
'max_doc_len': 1000
```

Can you give me tips about your paper?

1. How did you construct your training set based on the dataset?

In paper, you said you used queries from 6Min query logs and documents from Clueweb and Robust.
Can you give me more information about how did you construct (Query, Document1, Document2, Score) pairs?
Such as rankings of documents in training set and others 

2. Can you share the best performance parameter setting that you used to produce the performance in your paper?

Even though the best parameter seeting for my own dataset will be different, it can be a good start points for me to start tunning the model.

Thank you

**Comment by hamed-zamani** [comment]
Hi, thanks for your interest. 

1. The training data is constructed using weak supervision as was clearly described in our earlier work published at SIGIR 2017: https://ciir-publications.cs.umass.edu/pub/web/getpdf.php?id=1266
In fact, we use a term matching retrieval model (QL or BM25) to retrieve documents for each unique AOL query. We create all possible pairs from the retrieved result list and used the retrieval score to generate labels. In other words, you can assume that the order generated by the term matching retrieval method is correct.

2. Unfortunately, I don't have the learned parameters and the configuration file anymore (I am not at UMass anymore). But as far as I remember, the regularization term was much smaller than what you mentioned in your post. I think it was something close to 1e-7 if I am not mistaken. High value of this parameter leads to super sparse representations very fast and prevents the model from learning proper representations for capturing relevance.

I hope this helps.
Best,
Hamed

**Comment by tae1yoon** [comment]
Thank you for reply.
